### ISSUE TITLE
History of all rounds

### ISSUE THREAD

**Issue by Ayush11991** [rubyatmidnight/crashmultiplierstreaks#1]
Hey Ruby,

Can you explain how we can fetch history of all rounds till date, I believe it should be greater than 6 million rounds, but there must be a way to find out how many rounds from the Hashed one to the very first round of this game, isn't it?

Please share your inputs to the same. I tried putting 1000000000 like an infinite number but the command prompt just freezes.

Regards

**Comment by rubyatmidnight** [comment]
> Hey Ruby,
> 
> Can you explain how we can fetch history of all rounds till date, I believe it should be greater than 6 million rounds, but there must be a way to find out how many rounds from the Hashed one to the very first round of this game, isn't it?
> 
> Please share your inputs to the same. I tried putting 1000000000 like an infinite number but the command prompt just freezes.
> 
> Regards

sorry for the late reply.

There are only 6 million rounds for stake.com so far, but 10 million are planned. The way the seed works for crash is that the client seed is the previous seed, hashed. So if you encrypt the next round's seed, it should result be the current round's seed. This is the same method I use to generate a lot of seed pairs quickly, I just have a starting value, and then I hash each hashed result. 

The result is that we only have a history, and can't look into the future, unless we can break the current round's hash, or somehow see networking leaks/issues like has happened to other sites in the past, but Stake is pretty good about those things. 

There can technically be infinite rounds. But they have an end point, to 'prove' that they will all hash into that end point, and prove they didn't change anything along the way. 

They pre-made 10 million rounds, but we can only access as many as have been completed- otherwise, we can't guess at the round hashes. Without having the 10 millionth hash, we can't check all the previous ones. 

`"To prove our fairness we have generated a chain of 10,000,000 SHA256 hashes where each hash is the hash of the hexadecimal representation of the previous hash. The last hash in the chain is: 78a9757d3be42b74a3f70239078ad9317125fe9ee630d5bdada46de963e56752"`

source: https://bitcointalk.org/index.php?topic=5162888.0

The server seed (the static seed value, and not the round hash) was determined by saying, in advance, they would use the hash for this btc block: 

https://www.blockchain.com/btc/block-height/584500
or: `0000000000000000001b34dc6a1e86083f95500b096231436e9b25cbdd0075c4`

in short: there are literally only as many rounds as there have been completed that we can check and 'verify'. There are no more than 6 million. At about 140 rounds per hour, = 42k hours of gameplay, 1785 days, or almost 5 years of constant crash rounds. (July 08, 2019 was the date of that post, so that checks out perfectly.) 
If you want to generate more rounds, limbo is the same game. Crash is limbo, with early cashouts, and a shared seed. Any rounds generated 'before' 0 are going to be 'negative' I guess, 


Also, python will have a very hard time if you're trying to check for sequential values and it's a very large amount. This is not a well optimized program for huge numbers. 

As for dates, I can estimate a date and try to translate that into a round number. But it won't be exact.
